repair/row_level.cc

```diff
@@ -1383,14 +1393,57 @@ class repair_meta {
             // and _working_row_buf are ordered.
             utils::merge_to_gently(_working_row_buf, row_diff,
                  [this] (const repair_row& x, const repair_row& y) { return _cmp(x.boundary(), y.boundary()) < 0; });
+            for (auto& r : row_diff) {
+                thread::maybe_yield();
+                _working_row_buf_combined_hash.add(r.hash());
```

Comment:
I'm afraid I don't understand what this patch is doing.
If we (the master) get modifications from multiple remote nodes, how do we "combine" these modifications? What's the purpose of this "combined hash", and how can we "add" the hash of one diff - we may, for example, get the same missing column from two different remote nodes, we can't add them twice...
I'm probably missing something on how the new algorithm works.

Reply:
The combined hash is a hash of hashes in _working_row_buf. It is used to check if the rows inside _working_row_buf are identical. 

The new algorithm will not get the same row from more than one node.

Reply:
> The new algorithm will not get the same row from more than one node.

Why is it ok? Each node may contain different writes to the same row and repair should merge all of them.

Reply:
same row = same row with same content

Reply:
I don't get it. Followers can have a given row (identified by primary key) with different content. How can you avoid merging the rows from multiple followers if you direct them into a single fragment stream? Rows cannot repeat in the stream.